lacework/resource_lacework_query.go

```diff
@@ -0,0 +1,163 @@
+package lacework
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/lacework/go-sdk/api"
+)
+
+func resourceLaceworkQuery() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLaceworkQueryCreate,
+		Read:   resourceLaceworkQueryRead,
+		Update: resourceLaceworkQueryUpdate,
+		Delete: resourceLaceworkQueryDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: importLaceworkQuery,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"query_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The id of the query",
+			},
+			"query": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The query string",
+			},
+			"evaluator_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The query evaluator id",
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return old == "<<IMPLICIT>>" || new == "<<IMPLICIT>>"
+				},
+			},
+			"updated_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"result_schema": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceLaceworkQueryCreate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+	)
+
+	query := api.NewQuery{
+		QueryID:     d.Get("query_id").(string),
+		QueryText:   d.Get("query").(string),
+		EvaluatorID: d.Get("evaluator_id").(string),
+	}
+
+	log.Printf("[INFO] Creating Query with data:\n%+v\n", query)
+	response, err := lacework.V2.Query.Create(query)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.QueryID)
+	d.Set("owner", response.Data.Owner)
+	d.Set("updated_time", response.Data.LastUpdateTime)
+	d.Set("updated_by", response.Data.LastUpdateUser)
+	d.Set("result_schema", response.Data.ResultSchema)
+
+	log.Printf("[INFO] Created Query with guid %s\n", response.Data.QueryID)
+	return nil
+}
+
+func resourceLaceworkQueryRead(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+	)
+
+	log.Printf("[INFO] Reading Query with guid %s\n", d.Id())
+	response, err := lacework.V2.Query.Get(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.QueryID)
```

Comment:
I also don't think that Read should set the ID. I just went and looked and other resources in our codebase have this pattern but it is wrong.
```suggestion
```